spiceaidocs/docs/data-connectors/graphql.md

```diff
@@ -0,0 +1,78 @@
+---
+title: 'GraphQL Data Connector'
+sidebar_label: 'GraphQL Data Connector'
+description: 'GraphQL Data Connector Documentation'
+---
+
+## Federated SQL query
+
+To connect to any GraphQL endpoint as connector for federated SQL query, specify `graphql` as the selector in the `from` value for the dataset.
+
+```yaml
+datasets:
+  - from: graphql:your-graphql-endpoint
+    name: my_dataset
+    params:
+      json_path: data.some.nodes
+      query: |
+        {
+          some {
+            nodes {
+              field1
+              field2
+            }
+          }
+        }
+```
+
+## Configuration
+
+The GraphQL data connector can be configured by providing the following `params`:
+
+- `auth_token`: The authentication token to use to connect to the GraphQL server. Uses bearer authentication. E.g. `auth_token: github_pat_...`
+- `auth_token_key`: The secret key containing the authentication token to use to connect to the GraphQL server. Can be used instead of `auto_token`.
```

Comment:
```suggestion
- `auth_token_key`: The secret key containing the authentication token to use to connect to the GraphQL server. Can be used instead of `auth_token`.
```